src/assets/TypeScriptAsset.js

```diff
@@ -0,0 +1,34 @@
+const JSAsset = require('./JSAsset');
+const config = require('../utils/config');
+const localRequire = require('../utils/localRequire');
+
+class TypeScriptAsset extends JSAsset {
+  async transform() {
+    this.ast = await this.parse(this.contents);
+    this.isAstDirty = true;
+  }
```

Comment:
This method shouldn't be necessary, and overriding it here disables other transforms like babel (post typescript) if needed, and uglify. `parse` should already be called by the super `Asset` class.

Reply:
At first i tried without overwriting transform and it didn't parse the code. I'll try look into it a bit more

Reply:
In my latest addition to this PR i tried to implement it in a better way, hope you find this to be the correct way